caddyhttp/gzip/setup.go

```diff
@@ -106,6 +106,8 @@ func gzipParse(c *caddy.Controller) ([]Config, error) {
 			config.RequestFilters = append(config.RequestFilters, DefaultExtFilter())
 		}
 
+		config.ResponseFilters = append(config.ResponseFilters, SkipGzipped{})
```

Comment:
This is extremely important now that the Accept-Encoding header stays in the request, so add a test to ensure this response filter is always added.